### PR TITLE
Impose more strict type checking

### PR DESCRIPTION
**Pull request by skyw**
Added torch as additional_dependencies to mypy in pre-commit config.

Enabled error code explicit-override to mypy.

Fixed type issues based on AI's recommendation.

### CODE REVIEW

File: emerging_optimizers/riemannian_optimizers/normalized_optimizer.py [hunk 1]
```diff
@@ -65,8 +71,15 @@ def __init__(
...
+    @overload
+    def step(self, closure: None = ...) -> None: ...
+
+    @overload
+    def step(self, closure: Callable[[], float]) -> float: ...
+
     @torch.no_grad()  # type: ignore[misc]
-    def step(self, closure: Callable[[], float] | None = None) -> float | None:
+    @override
+    def step(self, closure: None = None) -> float | None:
```
[the remark this answers is not in the document]
Comment (skyw) [reply emerging_optimizers/riemannian_optimizers/normalized_optimizer.py]: fixed

File: emerging_optimizers/utils/modules.py [hunk 2]
```diff
@@ -108,6 +114,12 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
...
         return F.conv1d(x, weight, bias, self.stride, self.padding, self.dilation, self.groups)
 
+    @override
     def extra_repr(self) -> str:
+        if self.has_bias:
+            # trick extra_repr() to print bias=True
+            self.bias = torch.tensor(0)
         base_repr = super().extra_repr()
+        if self.has_bias:
+            del self.bias
```
[the remark this answers is not in the document]
Comment (skyw) [reply emerging_optimizers/utils/modules.py]: this looks ugly. I changed to plain string format.

File: emerging_optimizers/utils/modules.py [hunk 3]
```diff
@@ -108,6 +114,20 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
 
         return F.conv1d(x, weight, bias, self.stride, self.padding, self.dilation, self.groups)
 
+    @override
     def extra_repr(self) -> str:
-        base_repr = super().extra_repr()
-        return f"{base_repr}, flattened_param_shape={tuple(self.weight.shape)}"
+        s = "{in_channels}, {out_channels}, kernel_size={kernel_size}, stride={stride}"
```
[the remark this answers is not in the document]
Comment (skyw) [reply emerging_optimizers/utils/modules.py]: fixed

**Review by BoxiangW — APPROVED** [review]
LGTM
